core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,649 @@
+    await this.page.waitForSelector(forButtonToBeEnabled);
+    await this.clickOn(introSubmitButton);
+    showMessage('Card name updated successfully.');
+  }
+
+  /**
+   * This function helps in updating exploration intro text.
+   */
+  async updateExplorationIntroText(Introtext: string): Promise<void> {
+    await this.clickOn('div.e2e-test-state-edit-content');
+    await this.clickOn(explorationIntroBox);
+    await this.type(explorationIntroBox, Introtext);
+    await this.clickOn(introTitleSubmitButton);
+    showMessage('Intro text updated successfully.');
+  }
+
+  /**
+   * This function helps in adding interaction.
+   */
+  async addEndInteraction(): Promise<void> {
+    await this.page.waitForNavigation({waitUntil: 'networkidle0'});
+    await this.clickOn(interactionAddbutton);
+    await this.clickOn(endInteractionTab);
+    await this.clickOn(saveInteractionButton);
+    showMessage(
+      'End Exploration has been added and' +
+        'successfully created an exploration.'
+    );
+  }
+
+  /**
+   * This function helps in reaching setting tab successfully.
+   */
+  async goToSettingsTab(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      await this.page.waitForSelector(navBarOpener, {visible: true});
+      await this.clickOn(navBarOpener);
+      await this.clickOn(optionsDropDown);
+      await this.clickOn(mobileSettingsBar);
+      /**
+       * Open all dropdowns.
+       */
+      await this.clickOn(basicSettingsDropDown);
+      await this.clickOn(advanceSettingsDropDown);
+      await this.clickOn(rolesSettingsDropDown);
+      await this.clickOn(voiceArtistSettingsDropDown);
+      await this.clickOn(permissionSettingsDropDown);
+      await this.clickOn(feedbackSettingsDropdown);
+      await this.clickOn(explorationControlsSettingsDropDown);
+    } else {
+      await this.clickOn(settingsTab);
+    }
+  }
+
+  /**
+   * This function helps in updating Title.
+   */
+  async addTitle(Title: string): Promise<void> {
+    await this.page.waitForSelector(`${addTitleBar}:not([disabled])`);
+    await this.type(addTitleBar, Title);
+    showMessage('Title has been updated to ' + Title);
+  }
+
+  /**
+   * This function checks length of title bar at basic settings tab.
+   */
+  async expectTitleToHaveMaxLength(maxLength: number): Promise<void> {
+    const titleInput = await this.page.$('.e2e-test-exploration-title-input');
+    const title = await this.page.evaluate(input => input.value, titleInput);
+    const titleLength = title.length;
+
+    if (titleLength <= maxLength) {
+      showMessage(
+        'Title length is within the' +
+          ` allowed limit of ${maxLength} characters.`
+      );
+    } else {
+      throw new Error(
+        'Title length exceeds the allowed' +
+          ` limit of ${maxLength} characters.`
+      );
+    }
+  }
+
+  /**
+   * This function helps in adding a goal.
+   */
+  async updateGoalTo(goal: string): Promise<void> {
+    await this.clickOn(addGoal);
```

Comment:
addGoal sounds like a method. If you are referring to UI elements please use a format like this: xxxInputField, xxxButton, xxxLink, etc.